lib/modules/datasource/pypi/index.ts

```diff
@@ -43,45 +46,54 @@ export class PypiDatasource extends Datasource {
     packageName,
     registryUrl,
   }: GetReleasesConfig): Promise<ReleaseResult | null> {
-    let dependency: ReleaseResult | null = null;
     // TODO: null check (#22198)
-    const hostUrl = ensureTrailingSlash(
-      registryUrl!.replace('https://pypi.org/simple', 'https://pypi.org/pypi'),
-    );
+    const hostUrl = ensureTrailingSlash(registryUrl!);
+    const simpleHostUrl = hostUrl.replace(jsonPyPiUrl, simplePyPiUrl);
+    const pypiJsonHostUrl = hostUrl.replace(simplePyPiUrl, jsonPyPiUrl);
     const normalizedLookupName = normalizePythonDepName(packageName);
 
-    // not all simple indexes use this identifier, but most do
-    if (hostUrl.endsWith('/simple/') || hostUrl.endsWith('/+simple/')) {
+    const simpleDependencies = await this.getResultsViaSimple(
+      normalizedLookupName,
+      simpleHostUrl,
+    ).catch((err) => {
+      if (err.statusCode !== 404) {
+        throw err;
+      }
```

Comment:
@rarkins You had a [comment](https://github.com/renovatebot/renovate/pull/26505/files#r1591806441) that this might be too aggressive, what should this be instead? It's similar to current behaviour.

Reply:
It won't throw if a 404 is returned. I can add more error codes / remove this altogether, but I thought that's [undesirable](https://github.com/renovatebot/renovate/pull/26386#issuecomment-1865213796).

In general I think it would be weird to have the json api be available without the simple api also being available, since that's what's used by `pip`.

Reply:
If so, it's a breaking change so should be in a major release, not a fix/patch release. Even if you assume that all registries _should_ support simple, it's a change in Renovate if we now _only_ support that.

In my experience registries in the wild are surprisingly varying. Any time we've assumed "they all surely support X" it's turned out to be wrong.

Reply:
I don't have a preference one way or the other. I've removed the check for now.